src/tests/ftest/deployment/network_failure.py

```diff
@@ -0,0 +1,353 @@
+from job_manager_utils import get_job_manager
+
+
+class NetworkFailureTest(IorTestBase):
+    # pylint: disable=too-many-ancestors
+    """Test class Description: Verify network failure is properly handled and recovered.
+
+    :avocado: recursive
+    """
+    def __init__(self, *args, **kwargs):
+        """Store the info needed in tearDown."""
+        super().__init__(*args, **kwargs)
+        self.network_down_host = None
+
+    def tearDown(self):
+        """Bring ib0 back up in case the test crashed in the middle."""
+        self.log.debug("Call ip link set before tearDown.")
+        if self.network_down_host:
+            self.update_network_interface(
+                interface="ib0", state="up", host=self.network_down_host)
+        super().tearDown()
+
+    def run_ior_report_error(self, results, job_num, file_name, pool, container,
+                             namespace, timeout=None):
+        """Run IOR command and store the results to results dictionary.
+
+        Create a new IorCommand object instead of using the one in IorTestBase because
+        we'll run a test that runs multiple IOR processes at the same time.
+
+        Args:
+            results (dict): A dictionary object to store the ior metrics
+            job_num (int): Assigned job number
+            file_name (str): File name used for self.ior_cmd.test_file.
+            pool (TestPool): Pool to run IOR.
+            container (TestContainer): Container to run IOR.
+            namespace (str): IOR namespace.
+            timeout (int): Mpirun timeout value in sec. Defaults to None, in which case
+                infinite.
+        """
+        # Update the object class depending on the test case.
+        ior_cmd = IorCommand(namespace=namespace)
+        ior_cmd.get_params(self)
+
+        # Standard IOR prep sequence.
+        ior_cmd.set_daos_params(self.server_group, pool, container.uuid)
+        testfile = os.path.join("/", file_name)
+        ior_cmd.test_file.update(testfile)
+
+        manager = get_job_manager(
+            test=self, class_name="Mpirun", job=ior_cmd, subprocess=self.subprocess,
+            mpi_type="mpich", timeout=timeout)
+        manager.assign_hosts(
+            self.hostlist_clients, self.workdir, self.hostfile_clients_slots)
+
+        # Run the command.
+        try:
+            self.log.info("--- IOR command %d start ---", job_num)
+            ior_output = manager.run()
+            results[job_num] = [True]
+            # For debugging.
+            results[job_num].extend(IorCommand.get_ior_metrics(ior_output))
+            # Command worked, but append the error message if any.
+            results[job_num].append(ior_output.stderr_text)
+            self.log.info("--- IOR command %d end ---", job_num)
+        except CommandFailure as error:
+            self.log.info("--- IOR command %d failed ---", job_num)
+            results[job_num] = [False, "IOR failed: {}".format(error)]
```

Comment:
(pylint-consider-using-f-string) Formatting a regular string which could be a f-string